### PR TITLE
reduce it down a little bit

### PR DESCRIPTION
**Pull request by WildCryptoFox**

### CODE REVIEW

File: rust-toolchain [hunk 1]
```diff
@@ -1 +1 @@
-nightly-2019-11-24
+nightly-2019-11-26
```
Comment (WildCryptoFox) [inline rust-toolchain:1]: This is the current latest nightly. Which digresses from 2019-11-25 which appears to be happy with this code.

File: src/lib.rs [hunk 2]
```diff
@@ -1,184 +1,105 @@
...
-impl<T> Polygon<T> {
-    pub fn get_edge(&self, _index: usize) -> LineSegment2d<T> {
-        loop {}
-    }
-}
-
-fn problematic_function<Space>(_volume_space: &Space, material_surface: &Polygon<f64>)
-where
-    Space: GeometricFiniteElementSpace<f64, GeometryDim = U2, ReferenceDim = U2>,
+    Space: GeometricFiniteElementSpace<GeometryDim = U1/*, NodalDim = U1*/>,
```
Comment (WildCryptoFox) [inline src/lib.rs:65]: Simpler "fix" as below

File: src/lib.rs [hunk 3]
```diff
@@ -1,184 +1,105 @@
...
-    // Including this trait bound (which should already be satisfied)
-    // seems to resolve the ICE
-    //            + FiniteElementAllocator<f64, U2, U2, U2>
+    Space::Connectivity: ElementConnectivity<U1, NodalDim = Space::NodalDim>,
+    DefaultAllocator: FiniteElementAllocator<U1, Space::NodalDim>,
+    // Including this trait bound (which should already be satisfied) seems to resolve the ICE
+    //+ FiniteElementAllocator<U1, U1>
+    //
+    // This has the same effect as telling the compiler Space::NodalDim = U1; which was not already
+    // known to be satisfied.
```
Comment (WildCryptoFox) [inline src/lib.rs:76]: Or so it does in this case.

File: Cargo.toml [hunk 4]
```diff
@@ -3,8 +3,16 @@ name = "ice_testcase"
 version = "0.1.0"
 edition = "2018"
 
-[dependencies]
-nalgebra = { git = "https://github.com/rustsim/nalgebra.git", rev="31ef5f0ab02c6ecf279867f07cd63e16cece8b75", features = [ "serde-serialize" ] }
+[dependencies.nalgebra]
+git = "https://github.com/rustsim/nalgebra.git"
+rev = "31ef5f0ab02c6ecf279867f07cd63e16cece8b75"
+default-features = false
+features = ["serde-serialize", "default"]
```
Comment (WildCryptoFox) [inline Cargo.toml:10]: VERY weird that "default" acts differently to "std" here. Replacing this with "std" does not trigger the ICE.

File: src/lib.rs [hunk 5]
```diff
@@ -1,184 +1,85 @@
...
-    let _: Point2<f64> = material_surface_element.map_reference_coords(&eta).into();
+    let _: Point2<f64> = material_surface_element.map_reference_coords().into();
 
     // This is the source of another: they appear possibly to be due to the same bug,
     // but I can't say for certain, as certain manipulations of the above code only cause
     // one of them to fail, and not the other.
-    material_surface_element
-        .map_reference_coords(&eta)
-        .magnitude();
+    material_surface_element.map_reference_coords().magnitude();
```
Comment (WildCryptoFox) [inline src/lib.rs:84]: This is the same bug as above.

File: src/lib.rs (hunk 2 above)
Comment (Andlon) [reply src/lib.rs:65]: That is very strange, as the `Space` type has nothing to do with the the `FiniteElement` impls used in the function code.

**Comment by Andlon** [comment]
Thanks a lot @WildCryptoFox!

**Pull request merged**